niworkflows/utils/spaces.py

```diff
@@ -586,24 +595,28 @@ def get_fs_spaces(self):
         >>> SpatialReferences([
         ...     'fsnative',
         ...     'fsaverage6',
-        ...     Space(name='fsaverage', spec={'den': '30k'})
+        ...     Reference(space='fsaverage', spec={'den': '30k'})
         ... ]).get_fs_spaces()
         ['fsnative', 'fsaverage6']
 
         """
-        return [s.legacyname or s.name
-                for s in self._spaces
-                if s.legacyname or s.name == "fsnative"]
+        return [s.legacyname or s.space
+                for s in self.references
+                if s.legacyname or s.space == "fsnative"]
 
 
-class OutputSpacesAction(argparse.Action):
+class OutputReferencesAction(argparse.Action):
     """Parse spatial references."""
 
     def __call__(self, parser, namespace, values, option_string=None):
         """Execute parser."""
         spaces = getattr(namespace, self.dest) or SpatialReferences()
         for val in values:
-            for sp in Space.from_string(val):
+            val = val.rstrip(":")
+            # Should we support some sort of explicit "default" resolution?
```

Comment:
I don't think so, how would this handle volumes/surfaces? Doesn't an empty dict convey the notion of "default" everything?

Reply:
I am not confident this is the place for setting the default, but this would be necessary if we want (and I think we do) "default" and not-set different semantics.

From the user perspective, users should not be allowed to provide under-specified resampling targets. Therefore, setting a default would never harm.

From the point you added the checkpointing feature, these two different meanings of not-set and "default" are not as badly needed. But this could be helpful to only allow underspecified references for internal spaces.

Reply:
yeah, should probably go in `Reference.__iadd__`.

if you see it making life easier, i'm for it.

Reply:
Having `Reference.__iadd__` set a default will actually block the possibility of inserting spaces without spec (and we want to be able to do that)

Reply:
yeah, it would probably be in `check_specs`, I'll leave this for you to decide

Reply:
Let's not worry about this for now. I'll leave the comment there for when we come back to it.